src/items/associated-items.md

```diff
@@ -293,6 +293,25 @@ type that the definition has to implement.
 An *associated constant definition* defines a constant associated with a
 type. It is written the same as a [constant item].
 
+Unlike [free] constants, associated constant definitions undergo
+[constant evaluation] only when referenced.
```

Comment:
This seems odd, to make a guarantee by stating its *absence* on associated consts. Do we say explicitly anywhere that free constants are always evaluated, even when they are not used?

Reply:
> Do we say explicitly anywhere that free constants are always evaluated, even when they are not used?

I went through [Constant Evaluation](https://doc.rust-lang.org/stable/reference/const_eval.html) and [Constant items] and could not find it explicitly stated. Some possible ways forward:
 1. reword this line to say "Associated constant definitions undergo constant evaluation only when referenced.", removing the reference to free constants
 2. mention that unused free constants always undergo CTFE in [Constant items] and reword this sentence to link to it.

Do you have any preferences or recommendations?

[Constant items]:https://doc.rust-lang.org/stable/reference/items/constant-items.html

Reply:
2 sounds like something we will eventually want to do, but will probably require T-lang FCP to ensure that we are willing to commit to this as a guarantee.